src/hoverifier.ts

```diff
@@ -347,6 +356,18 @@ export type HoverProvider<C extends object, D> = (
     position: HoveredToken & C
 ) => Subscribable<MaybeLoadingResult<(HoverAttachment & D) | null>> | PromiseLike<(HoverAttachment & D) | null>
 
+/**
+ * Function that returns a Subscribable or PromiseLike of the ranges to be highlighted in the document.
+ * If a Subscribable is returned, it may emit more than once to update the content,
+ * and must indicate when it starts and stopped loading new content.
+ * It should emit a `null` result if the token has no highlights.
+ *
+ * @template C Extra context for the hovered token.
+ */
+export type DocumentHighlightProvider<C extends object> = (
+    position: HoveredToken & C
+) => Subscribable<MaybeLoadingResult<DocumentHighlight[] | null>> | PromiseLike<DocumentHighlight[] | null>
```

Comment:
Is loading state important in any way for document highlights? If not, I'd not use `MaybeLoadingResult` here.

Is there any semantic difference between `null` and empty array? If not, I'd not allow `null`